lib/lexer/src/lexer.rs

```diff
@@ -448,7 +454,7 @@ impl Lexer {
                 token.push(self.eat_char());
 
                 // Consume ('_'? digitbinary)*.
```

Comment:
Change occurences of `'_'?` to `'_'*`. Same applies for the `doc/grammar.peg` file